src/components/Dashboard/PendingTxsList/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { getTransactionQueue, TransactionSummary } from '@gnosis.pm/safe-react-gateway-sdk'
+import { List } from '@material-ui/core'
+import styled from 'styled-components'
+import { useSelector } from 'react-redux'
+
+import { isTransactionSummary } from 'src/logic/safe/store/models/types/gateway.d'
+import { grey400, sm } from 'src/theme/variables'
+import { GATEWAY_URL } from 'src/utils/constants'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { lastViewedSafe } from 'src/logic/currentSession/store/selectors'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import useOwnerSafes from 'src/logic/safe/hooks/useOwnerSafes'
+import { generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import { getChainById } from 'src/config'
+import PendingTxListItem from 'src/components/Dashboard/PendingTxListItem'
+
+const TransactionSkeleton = styled.div`
+  min-width: 270px;
+  height: 40px;
+  background-color: ${grey400};
+  border-radius: 8px;
+  margin: ${sm} auto;
+`
+
+const MAX_TXS_DISPLAY = 5
+
+const PendingTxsList = (): ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const defaultSafe = useSelector(lastViewedSafe)
+
+  const { hasLoaded: isLoadingOwnerSafes, ownerSafes } = useOwnerSafes()
+  const [loadingOwnerSafes, setLoadingOwnerSafes] = useState<boolean>()
+
+  // Selected Safe should come from the SideBar task when it's done
+  const [selectedSafe, setSelectedSafe] = useState<string | null>()
+
+  const [isFetchingPendingTxs, setIsFetchingPendingTxs] = useState<boolean>()
+  const [queuedTransactions, setQueuedTransactions] = useState<TransactionSummary[]>([])
+
+  useEffect(() => {
+    if (!isLoadingOwnerSafes) return
+
+    if (ownerSafes[chainId]?.length) {
+      setSelectedSafe(ownerSafes[chainId][0])
+    } else {
+      setSelectedSafe(defaultSafe)
+    }
+    setLoadingOwnerSafes(false)
+  }, [chainId, defaultSafe, isLoadingOwnerSafes, ownerSafes, selectedSafe])
+
+  // Fetch txs awaiting confirmations
+  useEffect(() => {
+    if (!selectedSafe) return
+
+    let isCurrent = true
+    const fetchQueuedTxs = async () => {
+      setIsFetchingPendingTxs(true)
+      const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(selectedSafe))
+      const txs = results
+        .filter(isTransactionSummary)
+        .slice(0, MAX_TXS_DISPLAY)
+        .map((item) => item.transaction)
+      if (isCurrent) {
+        setQueuedTransactions(txs)
+        setIsFetchingPendingTxs(false)
+      }
+    }
+    fetchQueuedTxs()
+
+    return () => {
+      isCurrent = false
+    }
+  }, [chainId, selectedSafe])
+
+  if (isFetchingPendingTxs) {
+    return (
+      <>
+        {Array.from(Array(MAX_TXS_DISPLAY).keys()).map((key) => (
+          <TransactionSkeleton key={key} />
+        ))}
+      </>
+    )
+  }
+
+  if (selectedSafe && !queuedTransactions.length && isFetchingPendingTxs === false) {
+    return <h3>This Safe has no pending transactions</h3>
```

Comment:
If there are pending transactions but they are already signed this message is shown which can be a bit misleading.

Reply:
We could probably show txns awaiting execution then. If everything is signed.

Reply:
That's not the current behaviour. We are showing the first 5 queued transactions.
If they are already totally signed it will show "2/2" for example. I've updated the PR description with new snapshots

Reply:
Perhaps _pending_ is not the best word choice

Reply:
'Queued' may make more sense. We would have to signify that there are more if there are >5 below though.

Reply:
changed the text to "queued"